marimbabot_planning/src/move_to_pose.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <ros/ros.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <tf2_ros/transform_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "marimba_move");
+    ros::NodeHandle node_handle;
+    ros::AsyncSpinner spinner(1);
+    spinner.start();
+
+    static const std::string PLANNING_GROUP = "arm_with_tcp";
+    moveit::planning_interface::MoveGroupInterface move_group_interface(PLANNING_GROUP);
+
+    const moveit::core::JointModelGroup* joint_model_group =
+        move_group_interface.getCurrentState()->getJointModelGroup(PLANNING_GROUP);
+
+
+    tf2_ros::Buffer tfBuffer;
+    tf2_ros::TransformListener tfListener(tfBuffer);
+    ros::Duration(1).sleep();
```

Comment:
Why do you sleep here? You can configure tf timeouts if this was an issue.

Reply:
without sleep, it got late and threw an error. I am not confident on TF functions so just put duration sleep which works.

Reply:
in principle a sleep after setting up a `TransformListener` is perfectly valid setup procedure.
TF lookup timeouts should represent the amount of time needed to wait during regular operation, and not the startup procedure.
In this concrete case it can be slightly more efficient to set the timeout below because the lookup is done only *once* right after constructing the `TransformListener`. But then again, the much better solution for a single loopup would be to use a [tf2_ros/buffer_server](https://github.com/ros/geometry2/blob/c92cd9a9073c255388599495d2f86799ecafa1bb/tf2_ros/CMakeLists.txt#L54-L63) and a single action.